utils/src/mesh.rs

```diff
@@ -33,7 +37,7 @@ pub trait KaboodlePeer {
 // End of tiny wrapper around Kaboodle.
 
 /// These are the roles we allow peers to advertise on the mesh
-#[derive(Debug, Clone, PartialEq, Eq, Decode, Encode, Deserialize, Serialize)]
+#[derive(Debug, Clone, PartialEq, Eq, Decode, Encode, Hash, Deserialize, Serialize)]
```

Comment:
All of the derive additions here are so we can stuff these things into a `HashMap`